### PR TITLE
New Extension: create-t3-app

### PR DESCRIPTION
**Pull request by mozzius**
## Description

Raycast interface for the [create-t3-app](https://github.com/t3-oss/create-t3-app) CLI tool for creating Next.js apps.

It's just a form that passes the options you choose through to the CLI tool. It opens a terminal window to do this (if there's a better way, please let me know - `child_process` wasn't working for me).

Requires npm, yarn, or pnpm to be installed (you pick which one to use in the preferences).
## Screenshot

That's pretty much it

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/create-t3-app/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          switch (value) {
+            case "ts":
+              setLangError(undefined);
+              break;
+            case "js":
+              setLangError("Wrong answer! Pick TypeScript!");
+              break;
+          }
+        }}
+      >
```
Comment (juhamust) [inline extensions/create-t3-app/src/index.tsx:154]: May I ask what is the purpose of this? Is it humour or unsupported feature? Why have the option at all? 🤔

Comment (mozzius) [reply extensions/create-t3-app/src/index.tsx:154]: Humor I guess? The CLI gives you the option to pick a language but forces you to pick JavaScript - I thought I would include it  in the raycast version to be faithful to that

File: extensions/create-t3-app/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} title="Create app (opens Terminal)" />
+          <Action onAction={() => openCommandPreferences()} title="Change extension preferences" />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField
+        id="appName"
+        title="Project name"
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:117]:
```suggestion
        title="Project Name"
```

File: extensions/create-t3-app/src/index.tsx (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/create-t3-app/src/index.tsx:154]: Agree, let's just remove it 😊

File: extensions/create-t3-app/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+              setLangError("Wrong answer! Pick TypeScript!");
+              break;
+          }
+        }}
+      >
+        <Form.Dropdown.Item value="ts" title="TypeScript" />
+        <Form.Dropdown.Item value="js" title="JavaScript" />
+      </Form.Dropdown>
+      <Form.Description text="Which packages would you like to enable?" />
+      <Form.Checkbox id="next-auth" label="next-auth" storeValue />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:159]:
```suggestion
      <Form.Checkbox id="next-auth" label="NextAuth " storeValue />
```

File: extensions/create-t3-app/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      <Form.Checkbox id="prisma" label="prisma" storeValue />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:160]:
```suggestion
      <Form.Checkbox id="prisma" label="Prisma" storeValue />
```

File: extensions/create-t3-app/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      <Form.Checkbox id="tailwind" label="tailwind" storeValue />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:161]:
```suggestion
      <Form.Checkbox id="tailwind" label="Tailwind" storeValue />
```

File: extensions/create-t3-app/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+      <Form.Checkbox id="trpc" label="trpc" storeValue />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:162]:
```suggestion
      <Form.Checkbox id="trpc" label="Trpc" storeValue />
```

File: extensions/create-t3-app/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        set shell to do script "cd ${folder} && ${command.join(" ")}"
+      end tell
+    `);
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} title="Create app (opens Terminal)" />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:110]:
```suggestion
          <Action.SubmitForm onSubmit={handleSubmit} title="Create App" />
```

File: extensions/create-t3-app/src/index.tsx [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+          <Action onAction={() => openCommandPreferences()} title="Change extension preferences" />
```
Comment (pernielsentikaer) [inline extensions/create-t3-app/src/index.tsx:111]:
```suggestion
          <Action onAction={() => openCommandPreferences()} title="Change Extension Preferences" />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/create-t3-app/src/index.tsx (hunk 6 above)
Comment (mozzius) [reply extensions/create-t3-app/src/index.tsx:162]: It's written 'tRPC', should that be respected? Or should 'Trpc' be used?

Comment (mozzius) [reply extensions/create-t3-app/src/index.tsx]: I went with 'tRPC'

**Comment by pernielsentikaer** [comment]
Sorry for the delay, checking again now 🧑‍🚒

**Review by pernielsentikaer — APPROVED** [review]
Hello 👋

I gave it a quick spin, works as expected, approved 🔥

**Pull request merged**